src/main/java/de/unistuttgart/iste/gits/flashcard_service/service/FlashcardService.java

```diff
@@ -0,0 +1,118 @@
+package de.unistuttgart.iste.gits.flashcard_service.service;
+
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardSetEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.dao.FlashcardSideEntity;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.mapper.FlashcardMapper;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.repository.FlashcardRepository;
+import de.unistuttgart.iste.gits.flashcard_service.persistence.repository.FlashcardSetRepository;
+import de.unistuttgart.iste.gits.flashcard_service.validation.FlashcardValidator;
+import de.unistuttgart.iste.gits.generated.dto.*;
+import jakarta.persistence.EntityNotFoundException;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+@Slf4j
+public class FlashcardService {
+
+    private final FlashcardRepository flashcardRepository;
+    private final FlashcardSetRepository flashcardSetRepository;
+    private final FlashcardMapper flashcardMapper;
+    private final FlashcardValidator flashcardValidator;
+
+
+    public Flashcard createFlashcard(CreateFlashcardInput flashcardInput) {
+        flashcardValidator.validateCreateFlashcardInput(flashcardInput);
+
+        FlashcardEntity mappedEntity = flashcardMapper.dtoToEntity(flashcardInput);
```

Comment:
TODO in this class: there is duplication in the  mapping logic. This should be extracted in methods or maybe done in the mapper, not in the service itsself.